test/e2e/test_gcp_broker.go

```diff
@@ -94,10 +98,54 @@ func SchedulerSourceWithGCPBrokerTestImpl(t *testing.T, authConfig lib.AuthConfi
 
 }
 
+// SmokeGCPBrokerTestImpl tests we can create a GCPBroker to ready state and we can delete a GCPBroker and its underlying resources.
+func SmokeGCPBrokerTestImpl(t *testing.T, authConfig lib.AuthConfig) {
+	client := lib.Setup(t, true, authConfig.WorkloadIdentity)
+	defer lib.TearDown(client)
+
+	brokerName := helpers.AppendRandomString("gcp")
+	// Create a new GCP Broker.
+	gcpBroker := client.CreateGCPBrokerV1Beta1OrFail(brokerName, knativegcptestresources.WithBrokerClassForBrokerV1Beta1(v1beta1.BrokerClass))
+
+	// Wait for broker ready.
+	client.Core.WaitForResourceReadyOrFail(brokerName, eventingtestlib.BrokerTypeMeta)
+
+	brokerresources.GenerateDecouplingTopicName(gcpBroker)
+
+	topicID := brokerresources.GenerateDecouplingTopicName(gcpBroker)
+	subID := brokerresources.GenerateDecouplingSubscriptionName(gcpBroker)
+
+	createdTopicExists := lib.TopicExists(t, topicID)
+	if !createdTopicExists {
+		t.Errorf("Expected topic%q to exist", topicID)
+	}
+
+	createdSubExists := lib.SubscriptionExists(t, subID)
+	if !createdSubExists {
+		t.Errorf("Expected subscription %q to exist", subID)
+	}
+
+	client.DeleteGCPBrokerOrFail(brokerName)
+	//Wait for 10 seconds for subscription to get deleted in gcp
+	time.Sleep(10 * time.Second)
```

Comment:
same here, if you can... Why was 20 the one above and this one 10?

Reply:
StackdriverSink for CloudAuditLogsSource and Scheduler Job for CloudSchedulerSource takes more time to create. That's why I don't use a constant. The wait time depends on the resources needed.

Reply:
I'm fine with it. Just do 2 * constant, and document why, so that other folks don't get confused